tests/monitoring/monitoring.go

```diff
@@ -637,8 +642,110 @@ var _ = Describe("[Serial][sig-monitoring]Prometheus Alerts", Serial, decorators
 			libwait.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
 		})
 	})
+
+	Context("System Alerts", func() {
+		AfterEach(func() {
+			waitUntilAlertDoesNotExist("KubeVirtNoAvailableNodesToRunVMs")
+		})
+
+		disableVirtHandler := func() *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{
+				Patches: []v1.CustomizeComponentsPatch{
+					{
+						ResourceName: virtHandler.deploymentName,
+						ResourceType: "DaemonSet",
+						Patch:        `{"spec":{"template":{"spec":{"nodeSelector":{"kubernetes.io/hostname":"does-not-exist"}}}}}`,
+						Type:         v1.StrategicMergePatchType,
+					},
+				},
+			}
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() string {
+				vh, err := virtClient.AppsV1().DaemonSets(originalKv.Namespace).Get(context.Background(), virtHandler.deploymentName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return vh.Spec.Template.Spec.NodeSelector["kubernetes.io/hostname"]
+			}, 90*time.Second, 5*time.Second).Should(Equal("does-not-exist"))
+
+			return kv
+		}
+
+		setEmulation := func(status bool) *v1.KubeVirt {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			kv.Spec.Configuration.DeveloperConfiguration.UseEmulation = status
+
+			kv, err = virtClient.KubeVirt(originalKv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() bool {
+				kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				return kv.Spec.Configuration.DeveloperConfiguration.UseEmulation
+			}, 90*time.Second, 5*time.Second).Should(Equal(status))
+
+			return kv
+		}
+
+		restoreVirtHandler := func(kv *v1.KubeVirt) {
+			originalKv := util.GetCurrentKv(virtClient)
+			kv, err := virtClient.KubeVirt(originalKv.Namespace).Get(originalKv.Name, &metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			kv.Spec.CustomizeComponents = v1.CustomizeComponents{}
+			kv, err = virtClient.KubeVirt(kv.Namespace).Update(kv)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		It("KubeVirtNoAvailableNodesToRunVMs should be triggered when there are no available nodes in the cluster to run VMs.", func() {
+			By("Scaling down virt-handler")
+			kv := disableVirtHandler()
+
+			By("Removing the kubevirt.io/schedulable label from all nodes")
+			nodes, err := virtClient.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			removeSchedulableLabelFromNodes(virtClient, nodes)
```

Comment:
I think the virt-handler should do this or?

Reply:
virt-handler usually takes care of this, but the nodes are schedulable, that's why I'm preventing virt-handler from running, so I can change the label manually